starlite/testing/test_client/client.py

```diff
@@ -104,6 +127,24 @@ def __init__(
             ),
         )
 
+    @property
+    def session(self) -> "CookieBackend":
+        warn_for_deprecated_session_interface()
```

Comment:
perhaps move the warning below the exception?

Reply:
I think it's helpful if the warning gets emitted even if the exception is hit, so if someone wants to fix this, they'll know the root cause immediately.

Reply:
ok, in general - these are non-blocking comments, so you can proceed as you see fit.